pkg/deps/context.go

```diff
@@ -0,0 +1,16 @@
+package deps
+
+import "context"
+
+type requestContainerContextKeyType struct{}
```

Comment:
Let's call the object that do dependency injection "Provider"; the object that do business logic "Service"; the object that coordinate different services "Flow"; the object that access data layer "Store".